mloop/learners.py

```diff
@@ -1299,6 +1312,15 @@ def fit_gaussian_process(self):
         self.scaled_costs = self.cost_scaler.fit_transform(self.all_costs[:,np.newaxis])[:,0]
         self.scaled_uncers = self.all_uncers / self.cost_scaler.scale_
         self.gaussian_process.set_params(alpha=self.scaled_uncers**2)
+        if self.cost_has_noise:
+            if np.isnan(self.noise_level):
+                # Set noise_level to its default value, which is the variance of
+                # the training data. This will only happen on first iteration
+                # since self.noise_level is overwritten.
+                self.noise_level = self.cost_scaler.var_
```

Comment:
I don't really have a sense of whether this is a sensible default. It seems to me as though it will massively overestimate the variance, but maybe the GP will quickly bring it back down to a reasonable value (I don't really know anything about GPs)?

Reply:
Yes it is very likely an overestimate, probably too large by roughly a factor of the SNR squared. I chose that default because it was equivalent to the previous default value of 1 (after scaling). It might make sense to set it to 0.1 or 0.01 times the variance instead.

Ideally the GP will bring it down to a more reasonable value, but that's not always the case. I've sometimes seen the issue discussed in section 1.7.2.1. [here](https://scikit-learn.org/stable/modules/gaussian_process.html#gaussian-process) in which the fitting routine fits the data as essentially a flat signal with large noise, rather than as a curvy signal with moderate noise. I'm thinking about another PR to deal with that issue more directly, but setting the default value here to something smaller might help. In particular it might encourage the fitting routine to converge to the right local minimum of hyperparameter values. i.e. the one that explains the data as a curvy signal with a bit of noise.

Reply:
Ah right, I hadn't thought about the default being equivalent to the old `1.0`, but yeah that makes sense. Thanks!